### ISSUE TITLE
Add Share Extension

### ISSUE THREAD

**Issue by jandamm** [jandamm/wallabag-macos#1]

**Comment by adamlogan** [comment]
Came here to ask for this. I'm really bummed this extension only shares within Safari, and not in other apps that use the Share Sheet on MacOS ( like RSS readers! ) I basically never touch Safari directly, but use RSS reader apps all the time that have share sheets.

**Comment by jandamm** [comment]
Thanks for the fee @adamlogan.

I'll take it into consideration.

**Comment by adamlogan** [comment]
It's the least I can do. Annual Apple dev fees is no joke. Thanks for considering this, it'd make your app valuable to more people and could help pick up more purchases.

I did figure out a workaround using a CLI tool + a shortcut with a shell script but it's still 3 × button presses per URL.

**Comment by jandamm** [comment]
@adamlogan I've added a Share Extension.
Please send a mail to speechy@jandamm.de if you (or others) want to try the beta.

Things to test would be
- if it works
- if the name in the sharing sheet is `Wallabag`*

\* I think it should say Wallabag but for me it shows `Share Extension` but I assume it's a caching error.

**Comment by adamlogan** [comment]
I overwrote the existing Wallabag Quick Save app with the Test Flight version 2026.3 (54). The Wallabag Quick Save extension works fine in Safari, but I'm not seeing Wallabag as an option in Share Sheets on MacOS under Extensions at:

General > Login Items & Extensions > Sharing (scroll all the way down close to bottom)

For what it's worth, I'm hanging back on Sequoia and am avoiding upgrading to Tahoma for now.

**Comment by jandamm** [comment]
@adamlogan sorry, my bad.
The deployment target of the extension was set to 26.2 (default) instead of 11.0.
I'll push a new beta with 11.0+ support.

**Comment by adamlogan** [comment]
Awesome.

Version 2026.3 (55) is working as an extension for the Share Sheet as expected. As far as I can tell, the loaded article does get sent over rather than just the URL as I have that option enabled.

- [x] It works
- [x] The extension properly shows as "Wallabag"


Articles get saved to Unread. It'd be nice if we had an option to set article as archived, add tags if desired. 

If you're up to it, you could mirror the options we get from the [Wallabager](https://wallabagger.wallabag.org/) extension/addon.

Wallabager is the most popular extension in the Firefox and Google Chrome store.

Functions the Wallabager extension has:

- Edit Wallabag Title of Article
- Set Article as archived in Wallabag
- Set Article as starred in Wallabag
- Remove Article from Wallabag (if on same page/article as already saved wallabag article)
- Text field for entering tags/labels.

Not sure if the API/URL building needs to be done before calling the share sheet or if these options can be implemented in the after clicking the share sheet button.

**Comment by jandamm** [comment]
No, the articles text isn't saved by the share sheet - only the link is sent and Wallabag fetches the content. So this wouldn't work for paywalled articles.
I'm not sure it would even be possible to get the content of the article via the share sheet. This is definitely dependent on the app which shares the content. The sharing app can decide what it shares. The extension can only check for a type of content (url, text, image, ...). But I wouldn't know if text would be the content, title or something completely different.

For now I want to keep the extensions simple and just save an article with one click and not having to fill out a mask. That's why it's called QuickSave.

**Comment by adamlogan** [comment]
I tested with [Current](https://www.currentreader.app/). It does fetch the article, not sure if it just sends a URL or the whole article when using the Share Sheet though. I would guess it is sending just the URL, although I do note the Wallabagger browser extension has a toggle to send the content of the article or maybe just tells Wallabag to fetch the whole article and provides the URL? 

Interesting re Wallabag. I find it odd I cannot find any setting in the webapp interface nor the Docker config about fetching articles. I'm guessing it's probably a config file inside the container and or default behavior. Not a high priority for me to figure out the exact mechanism at the moment.

99% of the time when I submit articles via web-browser, I am sending an article I have already read and marking it as archived so it doesn't show up in the Unread category which is my home screen on Wallabag. I would really appreciate it if you can save me from having to archive every article I save twice, once from the RSS Reader and then again in Wallabag. Perhaps there could be a setting in the Quick Save app to set the default category that the QuickSave Share function saves to.

**Comment by jandamm** [comment]
Maybe I can add a option where to safe new articles to.
You can create a new issue for that so others might chime in if they would like to have such an option too.

Closing this as the Share Extension is live in the AppStore now.
